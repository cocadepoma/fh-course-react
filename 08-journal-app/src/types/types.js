export const types = {
    login: "[Auth] Login",
    logout: "[Auth] Logout",

    uiSetError: "[UI] Set Error",
    uiRemoveError: "[UI] Remove Error",

    loginSetError: "[login] Set Error",
    loginRemoveError: "[login] Remove Error",
    loginStartLoading: "[login] Start Loading",
    loginFinishLoading: "[login] Finish Loading",

    notesAddNew: "[Notes] New note",
    notesActive: "[Notes] Set active note",
    notesLoad: "[Notes] Load notes",
    notesUpdated: "[Notes] Updated note",
    notesFileUrl: "[Notes] Updated image url",
    notesDelete: "[Notes] Delete note",
    notesLogoutCleaning: "[Notes] Logout Cleaning",
};
