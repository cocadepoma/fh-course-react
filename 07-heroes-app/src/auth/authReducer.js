import { types } from "../types/types";

export const authReducer = (state = {}, action) => {
    //Switch login/logout
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true,
            };
        case types.logout:
            return {
                logged: false,
            };
        default:
            return state;
    }
};
