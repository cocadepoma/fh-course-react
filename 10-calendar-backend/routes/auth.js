// const express = require("express");
// const router = express.Router;

/*
    Rutas de Usuarios / Auth
    host + /api/auth
*/

const { Router } = require("express");
// middleware de validaciones
const { check } = require("express-validator");
const { fieldValidator } = require("../middlewares/field-validators");
const { createUser, loginUser, renewToken } = require("../controllers/auth");
const { validatorJWT } = require("../middlewares/jwt-validators");

const router = Router();

// ruta, middlewares, funci√≥n
router.post(
    "/new",
    [
        check("name", "el nombre es obligatorio").not().isEmpty(),
        check("email", "el email es obligatorio").isEmail(),
        check("password", "el password debe de ser de 6 caracteres").isLength({ min: 6 }),
        fieldValidator,
    ],
    createUser
);

router.post(
    "/",
    [
        check("email", "el email es obligatorio").isEmail(),
        check("password", "el password debe de ser de 6 caracteres").isLength({ min: 6 }),
        fieldValidator,
    ],
    loginUser
);

router.get("/renew", validatorJWT, renewToken);

module.exports = router;
