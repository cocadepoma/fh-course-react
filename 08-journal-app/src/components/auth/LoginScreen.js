import React from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import { useDispatch, useSelector } from "react-redux";
import { startGoogleLogin, startLoginEmailPassword } from "../actions/auth";
import validator from "validator";
import { removeLoginError, setLoginError } from "../actions/login";

export const LoginScreen = () => {
    const dispatch = useDispatch();
    const { msgError, loading } = useSelector((state) => state.login);

    const [formValues, handleInputChange] = useForm({
        email: "nando@gmail.com",
        password: "123456",
    });

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        if (isLoginValid()) {
            dispatch(startLoginEmailPassword(email, password));
        }
    };

    const isLoginValid = () => {
        if (!validator.isEmail(email)) {
            dispatch(setLoginError("Email is not valid"));
            return false;
        }
        if (password.length < 6) {
            dispatch(setLoginError("Password length should be at least 6 characters"));
            return false;
        }
        dispatch(removeLoginError());
        return true;
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    return (
        <div className='animate__animated animate__fadeIn animate__fast'>
            <h3 className='auth__title'>Login</h3>
            <form onSubmit={handleLogin}>
                {msgError && <div className='auth__alert-error'>{msgError}</div>}
                <input
                    className='auth__input'
                    type='text'
                    placeholder='Email'
                    name='email'
                    autoComplete='off'
                    value={email}
                    onChange={handleInputChange}
                />
                <input
                    className='auth__input'
                    type='password'
                    placeholder='Password'
                    name='password'
                    value={password}
                    onChange={handleInputChange}
                />
                <button className='btn btn-primary btn-block' disabled={loading} type='submit'>
                    Login
                </button>

                <div className='auth__social-networks'>
                    <p>Login with social networks</p>
                    <div className='google-btn' onClick={handleGoogleLogin}>
                        <div className='google-icon-wrapper'>
                            <img
                                className='google-icon'
                                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
                                alt='google button'
                            />
                        </div>
                        <p className='btn-text'>
                            <b>Sign in with google</b>
                        </p>
                    </div>
                    <Link className='mt-5 link' to='/auth/register'>
                        Create new account
                    </Link>
                </div>
            </form>
        </div>
    );
};
