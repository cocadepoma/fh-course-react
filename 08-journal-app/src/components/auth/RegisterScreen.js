import React from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import { removeError, setError } from "../actions/ui";
import { useDispatch, useSelector } from "react-redux";
import validator from "validator";
import { startRegisterWithEmailPasswordName } from "../actions/auth";

export const RegisterScreen = () => {
    const dispatch = useDispatch();
    const { msgError } = useSelector((state) => state.ui);

    const defaultUser = {
        name: "Hernando",
        email: "nando@gmail.com",
        password: "123456",
        password2: "123456",
    };

    const [formValues, handleInputChange] = useForm(defaultUser);

    const { name, email, password, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        if (isFormValid()) {
            dispatch(startRegisterWithEmailPasswordName(email, password, name));
        }
    };

    const isFormValid = () => {
        if (name.trim().length <= 2) {
            dispatch(setError("The name input should be filled with a valid name"));
            return false;
        } else if (!validator.isEmail(email)) {
            dispatch(setError("The email input should be filled with a valid email"));
            return false;
        } else if (password !== password2 || password.length < 6) {
            dispatch(setError("Passwords doesn't match, or password too short"));
            return false;
        }
        dispatch(removeError());
        return true;
    };
    return (
        <div className='animate__animated animate__fadeIn animate__fast'>
            <h3 className='auth__title'>Register</h3>
            <form onSubmit={handleRegister}>
                {msgError && <div className='auth__alert-error'>{msgError}</div>}

                <input
                    className='auth__input'
                    type='text'
                    placeholder='Name'
                    name='name'
                    autoComplete='off'
                    value={name}
                    onChange={handleInputChange}
                />
                <input
                    className='auth__input'
                    type='text'
                    placeholder='Email'
                    name='email'
                    autoComplete='off'
                    value={email}
                    onChange={handleInputChange}
                />

                <input
                    className='auth__input'
                    type='password'
                    placeholder='Password'
                    name='password'
                    value={password}
                    onChange={handleInputChange}
                />
                <input
                    className='auth__input'
                    type='password'
                    placeholder='Confirm password'
                    name='password2'
                    value={password2}
                    onChange={handleInputChange}
                />

                <button className='btn btn-primary btn-block' type='submit'>
                    Register
                </button>

                <Link className='mt-5 link' to='/auth/login'>
                    Already Registered?
                </Link>
            </form>
        </div>
    );
};
