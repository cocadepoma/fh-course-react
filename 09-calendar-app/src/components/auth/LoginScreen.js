import React from "react";
import { useForm } from "../../hooks/useForm";
import { useDispatch } from "react-redux";

import "./login.css";
import { authStartLogin, startRegister } from "../../actions/auth";
import Swal from "sweetalert2";

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formLoginValues, handleLoginInputChange] = useForm({
        lEmail: "",
        lPassword: "",
    });

    const [formRegisterValues, handleRegisterInputChange] = useForm({
        rName: '',
        rEmail: "",
        rPassword: "",
        rPassword2: ''
    });

    const { lEmail, lPassword } = formLoginValues;
    const { rName, rEmail, rPassword, rPassword2 } = formRegisterValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(authStartLogin(lEmail, lPassword));
    };

    const handleRegister = (e) => {
        e.preventDefault();

        if (rPassword !== rPassword2) {
            return Swal.fire('Error', 'Las contrase単as deben de ser iguales', 'error');
        }

        dispatch(startRegister(rEmail, rPassword, rName));
    };

    return (
        <div className='container login-container'>
            <div className='row'>
                <div className='col-md-6 login-form-1'>
                    <h3>Ingreso</h3>
                    <form onSubmit={handleLogin}>
                        <div className='mb-3'>
                            <input
                                type='email'
                                className='form-control'
                                placeholder='Correo'
                                name='lEmail'
                                onChange={handleLoginInputChange}
                                value={lEmail}
                            />
                        </div>
                        <div className='mb-3'>
                            <input
                                type='password'
                                className='form-control'
                                placeholder='Contrase単a'
                                name='lPassword'
                                onChange={handleLoginInputChange}
                                value={lPassword}
                            />
                        </div>
                        <div className='form-group'>
                            <input type='submit' className='btnSubmit' value='Login' />
                        </div>
                    </form>
                </div>

                <div className='col-md-6 login-form-2'>
                    <h3>Registro</h3>
                    <form onSubmit={handleRegister}>
                        <div className='mb-3'>
                            <input
                                type='text'
                                className='form-control'
                                placeholder='Nombre'
                                onChange={handleRegisterInputChange}
                                name="rName"
                                value={rName}
                            />
                        </div>
                        <div className='mb-3'>
                            <input
                                type='email'
                                className='form-control'
                                placeholder='Correo'
                                onChange={handleRegisterInputChange}
                                name="rEmail"
                                value={rEmail}
                            />
                        </div>
                        <div className='mb-3'>
                            <input
                                type='password'
                                className='form-control'
                                placeholder='Contrase単a'
                                onChange={handleRegisterInputChange}
                                name="rPassword"
                                value={rPassword}
                            />
                        </div>

                        <div className='mb-3'>
                            <input
                                type='password'
                                className='form-control'
                                placeholder='Repita la contrase単a'
                                onChange={handleRegisterInputChange}
                                name="rPassword2"
                                value={rPassword2}
                            />
                        </div>

                        <div className='form-group'>
                            <input type='submit' className='btnSubmit' value='Crear cuenta' />
                        </div>
                    </form>
                </div>
            </div>
        </div>
    );
};
