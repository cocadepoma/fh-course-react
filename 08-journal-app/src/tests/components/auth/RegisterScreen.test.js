import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import { RegisterScreen } from "../../../components/auth/RegisterScreen";

import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";
import { types } from "../../../types/types";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null,
    },
    login: {
        loading: false,
        msgError: null,
    },
};

let store = mockStore(initialState);
// Descomentar para valorar cosas asíncronas
//store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
);

describe("Pruebas en <RegisterScreen />", () => {
    test("debe de mostrar correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de hacer el dispatch de la acción respectiva", () => {
        const emailField = wrapper.find('input[name="email"]');
        emailField.simulate("change", {
            target: {
                value: "",
                name: "email",
            },
        });

        wrapper.find("form").simulate("submit", {
            preventDefault() {},
        });

        const actions = store.getActions();

        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: "The email input should be filled with a valid email",
        });
    });

    test("debe de mostrar la caja de alerta con el error", () => {
        const initialState = {
            auth: {},
            ui: {
                loading: false,
                msgError: "Email no es correcto",
            },
            login: {
                loading: false,
                msgError: null,
            },
        };

        const store = mockStore(initialState);

        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScreen />
                </MemoryRouter>
            </Provider>
        );

        expect(wrapper.find(".auth__alert-error").text().trim()).toBe(initialState.ui.msgError);
        expect(wrapper.find(".auth__alert-error").exists()).toBe(true);
    });
});
