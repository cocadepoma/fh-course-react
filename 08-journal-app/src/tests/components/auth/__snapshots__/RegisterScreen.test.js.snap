// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen /> debe de mostrar correctamente 1`] = `
<div
  className="animate__animated animate__fadeIn animate__fast"
>
  <h3
    className="auth__title"
  >
    Register
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Hernando"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="nando@gmail.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Register
    </button>
    <a
      className="mt-5 link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already Registered?
    </a>
  </form>
</div>
`;
